api/layer/layer.go

```diff
@@ -259,6 +285,72 @@ func (f MsgHandlerFunc) HandleMessage(ctx context.Context, msg *nats.Msg) error
 	return f(ctx, msg)
 }
 
+// NewEncryptionParams create new params to encrypt with provided key.
+func NewEncryptionParams(key AES256Key) EncryptionParams {
+	return EncryptionParams{
+		enabled:     true,
+		customerKey: key,
+	}
+}
+
+// Key returns encryption key as slice.
+func (p EncryptionParams) Key() []byte {
+	return p.customerKey[:]
```

Comment:
I got why `EncryptionParams` has unexported fields, but can't we store `customerKey` as `[]byte`? 

As for me, type `AES256Key [32]byte` is redundant, and we cast it to a slice in all usages anyway.  And then we can make the fileds exported and drop methods. and `NewEncryptionParams` function.